documentation/src/docs/wire-protocol.md

```diff
@@ -20,54 +20,94 @@ All the requests and replies have 8 byte headers with two fields (all data is wr
-3. `suggestedLength` of Reply: Integer (4 bytes). The clients can request for the required length to the server (but the server may allot a different number of bytes.
-4. `delegationToken`: String (2 byte) followed by that many bytes of Java's Modified UTF-8. This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
+| **Field**   |**Datatype**   | **Description**     |
+|-------------|------------|----------|
+|  `Segment`| String| This Segment indicates the Stream Segment that was read. |
+| `Offset`   | Long| The `Offset` in the Stream Segment to read from. |
+| `suggestedLength` of Reply|Integer|The clients can request for the required length to the server (but the server may allot a different number of bytes.|
+|`delegationToken`|String| This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.|
+|`RequestId`| Long| This field contains the client-generated _ID_ that has been propagated to identify a client request.|
 
 
 ## Segment Read - Reply
 
-1. `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8. This Segment indicates the Stream Segment that was read.
-2. `Offset`: Long (8 bytes). The `Offset` in the Stream Segment to read from.
-3. `Tail`: Boolean (1 bit). If the read was performed at the tail of the Stream Segment.
-4. `EndOfSegment`: Boolean (1 bit). If the read was performed at the end of the Stream Segment.
-5. `Data`: Binary (remaining length in the message).
+| **Field**      | **Datatype**| **Description**     |
+|-------------|----------|--------|
+| `Segment`|String| This Segment indicates the Stream Segment that was read.|
+|`Offset`|Long| The `Offset` in the Stream Segment to read from.|
+|`Tail`|Boolean| If the read was performed at the tail of the Stream Segment.|
+| `EndOfSegment`| Boolean| If the read was performed at the end of the Stream Segment.|
+| `Data`| Binary| Remaining length in the message.|
+|`RequestId`| Long| This field contains the client-generated _ID_ that has been propagated to identify a client request.|
 
 The client requests to read from a particular Segment at a particular `Offset`. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server may decide transferring to the client more or less data than it was asked for, splitting that data in a suitable number of reply messages.
 
+### Segment API
+
+| **API**      | **Description**     |
+|-------------|----------|
+|`SegmentIsSealed`|The requested Segment is Sealed.|
+|`SegmentIsTruncated`|`Start offset`: Represents the offset at which the Segment is Truncated.|
+|`SegmentAlreadyExists`|The requested Segment already exists.|
+|`NoSuchSegment`| The requested Segment do not exist.|
+|`TableSegmentNotEmpty`||
+
```

Comment:
@tkaitchuck Provide the definition for `TableSegmentnotempty`

Reply:
It means that the table segment is not empty. (This is an error, for commands that only work on empty table segments)

Reply:
@tkaitchuck Accepted suggestion